### PR TITLE
[ENH] `hcrystalball` forecaster adapter

### PR DESCRIPTION
**Pull request by fkiraly**
This PR adds back @MichalChromcak's framework adapter to `hcrystalball`.

Since its removal, internal discussions have clarified inclusion policy of estimators in `sktime`, see here: https://www.sktime.org/en/stable/get_involved/governance.html#algorithm-inclusion-guidelines

The consolidated policy sees `sktime` as a library, to which any estimator can be contributed as long as minimal requirements on documentation are met.

I.e., perceived duplication, performance, relevance, personal taste of core developers, are not criteria we have decided we want to use going forward to decide on whether an algorithm should be included.

This PR moves the estimator into a new module `forecasting.adapters`, a location where potentially further adapters can be placed, say, to `gluonts`, `darts`, `pytorch-forecast` and so on (should someone implement those).

`hcrystalball` is not added as soft dependency in `all_extras`, which is fine, as `sktime` now also has full capability for dependency isolation.

### CODE REVIEW

File: sktime/forecasting/adapters/_hcrystalball.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+__author__ = ["MichalChromcak"]
+
+import pandas as pd
+from sklearn.base import clone
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+_check_soft_dependencies("hcrystalball", severity="warning")
```
Comment (aiwalter) [inline sktime/forecasting/adapters/_hcrystalball.py:14]: This is not in `all_extras`

**Review by aiwalter — CHANGES_REQUESTED** [review]
There was a core dev vote not to have this in past

**Comment by fkiraly** [comment]
> There was a core dev vote not to have this in past

* that was before the change in estimator inclusion policy
* which followed the discussion on HIVE-COTE where thought, @aiwalter, that it seemed to be a code of conduct violation to remove someone else's estimator

Same rules for everyone, I would suggest.

File: sktime/forecasting/adapters/_hcrystalball.py (hunk 1 above)
Comment (fkiraly) [reply sktime/forecasting/adapters/_hcrystalball.py:14]: if you read the PR description, it says:

> `hcrystalball` is not added as soft dependency in `all_extras`, which is fine, as `sktime` now also has full capability for dependency isolation.

(fine in my opinion)

**Comment by fkiraly** [comment]
If you want to block this or downvote this, please specify, on which grounds.

**Comment by aiwalter** [comment]
Lazy consensus failed on this PR

**Comment by fkiraly** [comment]
@aiwalter, that is not a productive statement - it amounts to you saying "I cause the lazy consensus to fail, and my reason is that the lazy consensus fails".

I.e., you say you reject this "just because".

I kindly ask you to reconsider your position on this PR and contribute productively to a discussion.

**Comment by fkiraly** [comment]
no follow-up, and not a clear change request. Can be merged by standing policy.

**Pull request merged**